### PR TITLE
feat: add `assert/is-palindrome`

### LINKED ISSUE #6391: [RFC]: add `@stdlib/assert/is-palindrome`
### Description

This RFC proposes to add a new assert util to check if a string is palindrome or not. 

### Related Issues

No.

### Questions

No.

### Other

Reference: [Palindrome | Wikipedia](https://en.wikipedia.org/wiki/Palindrome)

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by therealharshit**
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves #6391 
## Description

> What is the purpose of this pull request?

This pull request:

-   add `assert/is-palindrome`

## Related Issues

> Does this pull request have any related issues?

This pull request has the following related issues:

-   #6391
## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

### AI Assistance

> When authoring the changes proposed in this PR, did you use any kind of AI assistance?

-   [x] Yes
-   [ ] No

If you answered "yes" above, how did you use AI assistance?

-   [x] Code generation (e.g., when writing an implementation or fixing a bug)
-   [x] Test/benchmark generation
-   [x] Documentation (including examples)
-   [ ] Research and understanding

#### Disclosure

> If you answered "yes" to using AI assistance, please provide a short disclosure indicating how you used AI assistance. This helps reviewers determine how much scrutiny to apply when reviewing your contribution. Example disclosures: "This PR was written primarily by Claude Code." or "I consulted ChatGPT to understand the codebase, but the proposed changes were fully authored manually by myself.".

{{TODO: add disclosure if applicable}}

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/assert/is-palindrome/lib/main.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	}
+	len = v.length;
+	if ( len <= 1 ) {
+		return true;
+	}
+
+	i = 0;
+	j = len - 1;
+	while ( i < j ) {
+		if ( v.charAt( i ) !== v.charAt( j ) ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-palindrome/lib/main.js]: I don't believe this works quite the way you expect when dealing with grapheme clusters. You are essentially assuming ASCII characters here, but strings can contain surrogate pairs. You need to instead split into grapheme clusters and then do something similar to what you do here, but iterating over array elements. See https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/string/split-grapheme-clusters.

You should also be sure to include corresponding test cases verifying proper Unicode handling.

**Review by kgryte — CHANGES_REQUESTED** [review]
Left an initial comment.

File: lib/node_modules/@stdlib/assert/is-palindrome/lib/main.js (hunk 1 above)
Comment (therealharshit) [reply lib/node_modules/@stdlib/assert/is-palindrome/lib/main.js]: Tried to fix in 66ac84590dc760bde1f415f65001ef9026ccd54c.

**Comment by therealharshit** [comment]
> Left an initial comment.

Made some changes, let me know what you think?
